src/components/booksForm.js

```diff
@@ -1,52 +1,84 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import { CreateBook } from '../actions';
 import Categories from '../helper/constants';
 
-const BooksForm = ({ createBook }) => {
-  let title;
-  let category;
-
-  return (
-    <form
-      onSubmit={e => {
-        e.preventDefault();
-        createBook(
-          {
-            id: parseInt(100000 * Math.random(), 10),
-            title: title.value,
-            category: category.value,
-          },
-        );
-      }}
-    >
-      <div>
-        <label htmlFor="txtTitle">
-          Title
-          <input type="text" id="txtTitle" name="txtTitle" ref={input => { (title = input); }} />
-        </label>
-      </div>
-      <div>
-        <label htmlFor="selCategory">
-          Category
-          <select name="selCategory" id="selCategory" ref={select => { (category = select); }}>
-            {
-              Categories.map(item => (
-                <option key={item} value={item}>{item}</option>
-              ))
-            }
-          </select>
-        </label>
-      </div>
-      <div>
-        <button type="submit">
-          Add Book
-        </button>
-      </div>
-    </form>
-  );
-};
+class BooksForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.title = '';
```

Comment:
* [ ] If we want to use reference let make sure to initialize it as the docs suggest, ie `this.title = React.createRef();`

Reply:
Kindly reset this to what it was,
 `
this.state = {
      title: " ",
      category: " ",
    };
`
I was referring to `this.title` and `this.category` variable, since they are referring to input and you would like to get their value by reference you should use `React.createRef()`, it implies having `this.title = React.createRef()` and `this.category = React.createRef()`. What you have done is ok, i sought make initialize the variables with the `React.createRef()` instead of empty string. 🤗